src/stores/commandStateStore.ts

```diff
@@ -20,12 +25,19 @@ export const resetCommandState = () => {
 
 /** Updates the command state to the current state of the document. */
 export const updateCommandState = () => {
-  commandStateStore.update({
-    bold: document.queryCommandState('bold'),
-    italic: document.queryCommandState('italic'),
-    underline: document.queryCommandState('underline'),
-    strikethrough: document.queryCommandState('strikethrough'),
-  })
+  const state = store.getState()
+  if (!state.cursor) return
+  const thought = pathToThought(state, state.cursor)
```

Comment:
Inline `pathToThought(state, state.cursor)` below so that it is only called when `isActive` is false.

Normally I wouldn't worry about this minor optimization, but `updateCommandState` is called on every keystroke, so it is highly performance sensitive.